composer/utils/eval_client/eval_client.py

```diff
@@ -7,28 +7,31 @@
 """Abstract class for utilities that access and run code on serverless eval clients."""
 
 import abc
-from typing import Dict
+from typing import Dict, List
 
 __all__ = ['EvalClient']
 
 
 class EvalClient(abc.ABC):
     """Abstract class for implementing eval clients, such as LambdaEvalClient."""
 
-    def invoke(self, payload: Dict[str, str]) -> bool:
-        """Invoke a provided dictionary payload to the client.
+    def invoke(self, payload: List[List[List[Dict[str, str]]]]) -> List[List[List[bool]]]:
+        """Invoke a provided batch of dictionary payload to the client.
 
-        For code generation, the payload is a JSON with the following attributes:
+        For code generation, the payload is a list of list of lists of JSONs with the following attributes:
```

Comment:
The nesting structure should be described here. Why does it accept a list of list of lists? Please detail:
1. the breakdown (prompt group, test case, etc)
2. specify that the eval client is agnostic to this and should just iterate over each payload maintaining the parent list structure

Reply:
done